common/sema.h

```diff
@@ -203,8 +203,14 @@ class LightweightSemaphore
 
     void wait()
     {
-        if (!tryWait())
-            waitWithPartialSpinning();
+        if (!tryWait()) {
+            // We will allow spinning only if we know multiple "hardware thread contexts" are available
+            if (std::thread::hardware_concurrency > 1) {
```

Comment:
Looks like `std::thread::hardware_concurrency` [is actually a function](http://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency)